lib/util/finally-mixin.js

```diff
@@ -0,0 +1,10 @@
+const noop = require('./noop');
+
+const finallyMixin = (prototype) =>
+  Object.assign(prototype, {
+    finally(onFinally) {
+      return this.then().finally(onFinally);
+    },
+  });
+
+module.exports = Promise.prototype.finally ? noop : finallyMixin;
```

Comment:
We should probably leave a comment here to explain why this logic is conditional.

```suggestion
// FYI: Support for `Promise.prototype.finally` was not introduced until Node 9.
//        Therefore, Knex will need to conditionally inject support for `.finally(..)`
//        until support for Node 8 is officially dropped.
module.exports = Promise.prototype.finally ? noop : finallyMixin;
```